spyder/widgets/projects/tests/test_project_explorer.py

```diff
@@ -7,26 +7,59 @@
 """
 Tests for explorer.py
 """
+# Standard imports
+import os
+import os.path as osp
 
 # Test library imports
 import pytest
 
 # Local imports
 from spyder.widgets.projects.explorer import ProjectExplorerTest
 
+
 @pytest.fixture
-def setup_projects_explorer(qtbot):
-    """Set up ProjectExplorerWidgetTest."""
-    project_explorer = ProjectExplorerTest()
+def project_explorer(qtbot, request, tmpdir):
+    """Setup Project Explorer widget."""
+    directory = request.node.get_marker('change_directory')
+    if directory:
+        project_dir = str(tmpdir.mkdir('project'))
+    else:
+        project_dir = None
+    project_explorer = ProjectExplorerTest(directory=project_dir)
     qtbot.addWidget(project_explorer)
-    return project_explorer
+    return (project_explorer, project_dir)
+
+
+@pytest.mark.change_directory
+def test_change_directory_in_project_explorer(project_explorer, qtbot):
+    """Test changing a file from directory in the Project explorer."""
+    # Create project
+    project, project_dir = project_explorer
```

Comment:
This needs to be now

    project = project_explorer
    project_dir = project.directory